grakn-grpc/src/main/java/ai/grakn/grpc/ConceptProperty.java

```diff
@@ -0,0 +1,200 @@
+            val -> val.isAttribute() ? val.asAttribute().dataType() : val.asAttributeType().getDataType(),
+            val -> GrpcUtil.convert(val.getDataType()),
+            (builder, val) -> builder.setDataType(GrpcUtil.convert(val))
+    );
+
+    public static final ConceptProperty<Label> LABEL = create(
+            LabelProperty,
+            concept -> concept.asSchemaConcept().getLabel(),
+            val -> GrpcUtil.convert(val.getLabel()),
+            (builder, val) -> builder.setLabel(GrpcUtil.convert(val))
+    );
+
+    public static final ConceptProperty<Boolean> IS_IMPLICIT = create(
+            IsImplicit,
+            concept -> concept.asSchemaConcept().isImplicit(),
+            ConceptPropertyValue::getIsImplicit,
+            ConceptPropertyValue.Builder::setIsImplicit
+    );
+
+    public static final ConceptProperty<Boolean> IS_INFERRED = create(
+            IsInferred,
+            concept -> concept.asThing().isInferred(),
+            ConceptPropertyValue::getIsInferred,
+            ConceptPropertyValue.Builder::setIsInferred
+    );
+
+    public static final ConceptProperty<Boolean> IS_ABSTRACT = create(
+            IsAbstract,
+            concept -> concept.asType().isAbstract(),
+            ConceptPropertyValue::getIsInferred,
+            ConceptPropertyValue.Builder::setIsInferred
+    );
+
+    public static final ConceptProperty<Pattern> WHEN = create(
+            When,
+            concept -> concept.asRule().getWhen(),
+            val -> GrpcUtil.convert(val.getWhen()),
+            (builder, val) -> builder.setWhen(GrpcUtil.convert(val))
+    );
+
+    public static final ConceptProperty<Pattern> THEN = create(
+            Then,
+            concept -> concept.asRule().getThen(),
+            val -> GrpcUtil.convert(val.getThen()),
+            (builder, val) -> builder.setThen(GrpcUtil.convert(val))
+    );
+
+    public static final ConceptProperty<String> REGEX = create(
+            Regex,
+            concept -> concept.asAttributeType().getRegex(),
+            ConceptPropertyValue::getRegex,
+            ConceptPropertyValue.Builder::setRegex
+    );
+
+    public static ConceptProperty<?> fromGrpc(GraknOuterClass.ConceptProperty conceptProperty) {
+        switch (conceptProperty) {
+            case ValueProperty:
+                return VALUE;
+            case DataTypeProperty:
+                return DATA_TYPE;
+            case LabelProperty:
+                return LABEL;
+            case IsImplicit:
+                return IS_IMPLICIT;
+            case IsInferred:
+                return IS_INFERRED;
+            case IsAbstract:
+                return IS_ABSTRACT;
+            case When:
+                return WHEN;
+            case Then:
+                return THEN;
+            case Regex:
+                return REGEX;
+            default:
+            case UNRECOGNIZED:
+                throw new IllegalArgumentException("Unrecognised " + conceptProperty);
```

Comment:
does it makes sense to put the error message in `ErrorMessage` class (which is somewhere in `grakn-engine` if i am not mistaken)

Reply:
Maybe. This will only happen if the gRPC server you're connecting to is bugged - so it's kind of meant to be "unreachable".